File: commands/commands.py
import requests 
import json
import os

#Change the prompt to test different commands.
prompt = "rotate the model by 90 degrees"

def get_api_key():
    """Get OpenAI API key from environment variables."""
    return os.getenv('OPENAI_API_KEY')

def intentAPI(prompt, openAI_api_key):
    """Get the intent of a prompt """
    url = "https://intent.blockbuilders.host"
    headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    }
    data = {
        "prompt": prompt,
    }
    try:
        response = requests.post(url, headers=headers, data=json.dumps(data))
        response.raise_for_status()
    except requests.exceptions.RequestException as e:
        print(f"Request failed: {e}")
        return None

    try:
        output = json.loads(response.text)
        return output["intent"]
    except json.JSONDecodeError as e:
        print(f"Failed to decode JSON: {e}")
        return None

def commandAPI(prompt, intention, openAI_api_key):
    """Get command from OpenAI API."""
    intention = intentAPI(prompt, openAI_api_key)
    if intention is None:
        return None
        
    url = "https://commands.blockbuilders.host"
    headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    }
    data = {
        "prompt": prompt,
        "intent": intention,
    }
    try:
        response = requests.post(url, headers=headers, data=json.dumps(data))
        response.raise_for_status()
    except requests.exceptions.RequestException as e:
        print(f"Request failed: {e}")
        return None

    try:
        output = json.loads(response.text)
        return output["command"]
    except json.JSONDecodeError as e:
        print(f"Failed to decode JSON: {e}")
        return None

def buildcommandAPI(prompt, intention, openAI_api_key):
    """Get build command from OpenAI API."""
    url = "https://build.blockbuilders.host"
    headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    }
    data = {
        "prompt": prompt,
        "intent": intention,
    }
    try:
        response = requests.post(url, headers=headers, data=json.dumps(data))
        response.raise_for_status()
    except requests.exceptions.RequestException as e:
        print(f"Request failed: {e}")
        return None

    try:
        output = json.loads(response.text)
        return output
    except json.JSONDecodeError as e:
        print(f"Failed to decode JSON: {e}")
        return None

def main():
    """Main function."""
    openAI_api_key = get_api_key() 
    intention = intentAPI(prompt, openAI_api_key)
    if intention is None:
        return None
    elif intention in ["move", "define", "rotate", "scale", "replace", "fill"]:
        command = buildcommandAPI(prompt, intention, openAI_api_key)
    else:
        command = commandAPI(prompt, intention, openAI_api_key)
    if command is not None:
        print(command)
    
if __name__ == "__main__":
    main()

File: commands/commands.js
const axios = require('axios');
// Change the prompt to test different commands.
let prompt = "move the model to the left by 10 blocks";

function getApiKey() {
    // Get OpenAI API key from environment variables.
    return process.env.OPENAI_API_KEY;
}

//you'll need to run this npm install axios
async function intentAPI(prompt, openAI_api_key) {
    // Get the intent of a prompt
    let url = "https://intent.blockbuilders.host";
    let headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    };
    let data = {
        "prompt": prompt,
    };
    try {
        let response = await axios.post(url, data, {headers: headers});
        return response.data.intent;
    } catch (error) {
        console.log(`Request failed: ${error}`);
        return null;
    }
}

async function commandAPI(prompt, intention, openAI_api_key) {
    // Get command from OpenAI API.    
    let url = "https://commands.blockbuilders.host";
    let headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    };
    let data = {
        "prompt": prompt,
        "intent": intention,
    };
    try {
        let response = await axios.post(url, data, {headers: headers});
        return response.data.command;
    } catch (error) {
        console.log(`Request failed: ${error}`);
        return null;
    }
}
async function buildCommandAPI(prompt, intention, openAI_api_key) {
    // Get command from OpenAI API.    
    let url = "https://build.blockbuilders.host";
    let headers = {
        "openai-api-key": openAI_api_key,
        "Content-Type": "application/json",
    };
    let data = {
        "prompt": prompt,
        "intent": intention,
    };
    try {
        let response = await axios.post(url, data, {headers: headers});
        return response.data;
    } catch (error) {
        console.log(`Request failed: ${error}`);
        return null;
    }
}

async function main() {
    // Main function.
    let openAI_api_key = getApiKey();
    let intention = await intentAPI(prompt, openAI_api_key);
    console.log(intention);
    let command = null;
    if (intention === null) {
        return null;
    }
    else if (intention === "move" || intention === "define" || intention === "rotate" || intention === "scale" || intention === "replace" || intention === "fill") {
        command = await buildCommandAPI(prompt, intention, openAI_api_key);
    }
    else {
        command = await commandAPI(prompt, intention, openAI_api_key);
    }
    if (command !== null) {
        //Logs the command to the console. You can add your code here.
        console.log(command);
    }
}

main();